access_error.go

```diff
@@ -46,8 +46,17 @@ func (f *Fosite) writeJsonError(rw http.ResponseWriter, err error) {
 	if err != nil {
 		if f.SendDebugMessagesToClients {
 			// Poor man's JSON encoding. We do not want to use full JSON encoding because we just had an error.
+			// The characters that MUST be escaped: quotation mark, reverse solidus, and the control characters
+			// (U+0000 through U+001F).
+			// See: https://tools.ietf.org/html/std90#section-7
 			errorMessage := err.Error()
+			// Escape reverse solidus.
 			errorMessage = strings.ReplaceAll(errorMessage, `\`, `\\`)
+			// Escape control characters.
+			for r := rune(0); r < ' '; r++ {
+				errorMessage = strings.ReplaceAll(errorMessage, string(r), fmt.Sprintf(`\u%04d`, r))
+			}
+			// Escape quotation mark.
 			errorMessage = strings.ReplaceAll(errorMessage, `"`, `\"`)
```

Comment:
Maybe make this a small helper function to avoid code duplication and add a small test? :)

Reply:
Done.